src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.cpp

```diff
@@ -381,19 +381,15 @@ HeapWord* ShenandoahCardCluster::first_object_start(const size_t card_index, con
   oop obj = cast_to_oop(p);
   assert(oopDesc::is_oop(obj), "Should be an object");
 #ifdef ASSERT
```

Comment:
Minor: Move `#ifdef ASSERT` before `oop obj = ...` assignment. I am fairly certain that load would be folded in product configuration, but there is no point in risking it. Same with `#endif` below, move it just before return. Plus, change `guarantee(false, "...")` to `fatal("...")`.